### ISSUE TITLE
simple encoding options?

### ISSUE THREAD

**Issue by yyzguy** [dscottbuch/cTiVo#247]
Hello,

I'm new to cTivo and really impressed.  It's amazing simple to get shows of the Tivo.  Thanks!

My ultimate goal is to download a recorded concert and split it into separate files for each song of the concert in order to make a "video jukebox."   I'll probably use plex for the jukebox.  Seems that plex really likes mkv files.

So, I'm wondering if there's a way to either
1) have cTivo skip the encoding process and just leave the downloaded .mpg file.  With this option I would do the splitting and re-encoding myself with other tools.
or
2) have an encoding option that converts the downloaded show into .mkv using the native resolution and audio format of the original file.   So if the original was 1080i, then output to 1080i.

Thank you in advance

**Comment by yyzguy** [comment]
Nevermind...I figured it out.   Their were hidden encoders that were exactly what I needed.  

Cheers

**Comment by mackworth** [comment]
Glad to hear.  Btw, I don't think Plex does any better with MKV vs MP4.  Let me know if you find differently. Secondly, the Default Format should do what you describe in #2; again, let me know what you find different from that

**Comment by yyzguy** [comment]
Well, I'm not sure why I was preferring MKV over MP4 for Plex.  (I've been doing alot of experimenting and my brain is overloaded with various formats and such)

However, now I have run into a different (but related) issue.
It seems that plex media server needs to transcode the audio track from DTS to AAC in order to support plex client.   

I'm trying to reduce the amount of work that plex needs to do, so I'd like to have the audio track of Tivo downloaded files be converted AAC (instead of native, like I had originally posted).  I've been reading the ffmeg documentation and it's a bit overwhelming.  I'm sure there's already a setting in cTivo, but so far I haven't found it.

Suggestion?

**Comment by mackworth** [comment]
So for multichannel shows, the normal behavior for cTiVo nowadays is to take the AC3 that TiVo provides and have ffmpeg add a stereo AAC as well. So if you look inside the file, you'll see a 6-channel AC3 track and a 2-channel AAC track.  (See the [Format chart here](https://github.com/dscottbuch/cTiVo/wiki/Video-Formats).)  This seems to be fine for Plex.  I don't think TiVo produces any DTS?  

I would also note that Plex suggests "[don't do passthrough](https://support.plex.tv/hc/en-us/articles/212072167-Audio-Configuration-Guide-Plex-Media-Player)"

**Comment by yyzguy** [comment]
Giant Face Palm!

I have two versions of a particular show.  One was downloaded from Tivo using cTivo.  The other was ripped from Blu ray.   Indeed the Tivo version has both AAC and AC3.  It was the blu ray version where I saw "converting DTS to AAC".    Sorry for the confusion.

Also, for testing I am using the version of Plex client that is built-in to my Tivo.   My testing seems to suggest this particular plex client cannot handle AC3, because the server is showing AC3 being converted to AAC.   Obviously, for concerts AC3 would be preferred!

I have an Nvidia Shield on order and that’s what I will ultimately be using. I think the nvidia Shield, as well as other full clients can handle the AC3 without converting to AAC.  

Back to your earlier comment about MP4 vs MKV:  You are correct.  Plex can handle MP4 just as well as MKV.   

I’ve been working with MKV because I’ve been finding nice tools, such as MakeMKV for ripping blu ray and MKVToolNix for chapter splitting.

What I may end up doing, is using cTivo and MP4 for things I get from Tivo and MKV for Blu Ray.

I also noticed that I can often rename MP4 to MKV and vice versa.   I realize they are both containers, so my guess is as long as what they contain are compatible with both, the name itself doesn’t matter and the software takes care of whatever it needs to do.

I do have another question for you, but for that I will open a new issue.

Thank you so much for your time
